pkg/apis/pipeline/v1beta1/provenance_conversion.go

```diff
@@ -16,19 +16,26 @@ package v1beta1
 import (
 	"context"
 
+	"github.com/tektoncd/pipeline/pkg/apis/config"
 	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
 )
 
 func (p Provenance) convertTo(ctx context.Context, sink *v1.Provenance) {
-	new := v1.ConfigSource{}
-	p.ConfigSource.convertTo(ctx, &new)
-	sink.ConfigSource = &new
+	if p.ConfigSource != nil {
+		new := v1.ConfigSource{}
+		p.ConfigSource.convertTo(ctx, &new)
+		sink.ConfigSource = &new
+	}
+	sink.FeatureFlags = config.FromContextOrDefaults(ctx).FeatureFlags
```

Comment:
Could you explain why we are getting the feature flags from context here? Inside the conversion webhook, shouldn't we be getting the feature flag values from `p` and adding them to `sink` (like we are doing for `ConfigSource` instead of getting the data from the configMap?

Reply:
good catch. I now get it from `p` instead of the config map.